### PR TITLE
Update window bounds attributes on change

### PR DESCRIPTION
**Pull request by maddie**
This commit depends on PR asticode/astilectron#55, and targets astilectron release 0.53.0 (yet to be tagged and released).

With the added listener and function, users should now be able to get the window's position and size in real time.

Should fix issue #127

### CODE REVIEW

**Comment by maddie** [comment]
The pushed code depends on astilectron `0.53.0` and CI fails to download it, so it fails :(

File: window.go [hunk 1]
```diff
@@ -246,6 +248,30 @@ func newWindow(ctx context.Context, l astikit.SeverityLogger, o Options, p Paths
...
+
+	w.On(EventNameWindowEventDidFinishLoad, updateBoundsFunc(w))
+	w.On(EventNameWindowEventMaximize, updateBoundsFunc(w))
+	w.On(EventNameWindowEventMove, updateBoundsFunc(w))
+	w.On(EventNameWindowEventMoved, updateBoundsFunc(w))
+	w.On(EventNameWindowEventResize, updateBoundsFunc(w))
+	w.On(EventNameWindowEventResizeContent, updateBoundsFunc(w))
+	w.On(EventNameWindowEventUnmaximize, updateBoundsFunc(w))
+	w.On(EventNameWindowEventWillMove, updateBoundsFunc(w))
+
```
Comment (asticode) [inline window.go:274]: Can you also remove the portions of code that was updating window bounds before receiving a terminated event?

- https://github.com/asticode/go-astilectron/blob/master/window.go#L397
- https://github.com/asticode/go-astilectron/blob/master/window.go#L481
- https://github.com/asticode/go-astilectron/blob/master/window.go#L494
- https://github.com/asticode/go-astilectron/blob/master/window.go#L507

File: window_test.go [hunk 2]
```diff
@@ -65,6 +65,9 @@ func TestWindow_Actions(t *testing.T) {
 	var d = newDisplay(&DisplayOptions{Bounds: &RectangleOptions{PositionOptions: PositionOptions{X: astikit.IntPtr(1), Y: astikit.IntPtr(2)}, SizeOptions: SizeOptions{Height: astikit.IntPtr(1), Width: astikit.IntPtr(2)}}}, true)
 	testObjectAction(t, func() error { return w.MoveInDisplay(d, 3, 4) }, w.object, wrt, "{\"name\":\""+EventNameWindowCmdMove+"\",\"targetID\":\""+w.id+"\",\"windowOptions\":{\"x\":4,\"y\":6}}\n", EventNameWindowEventMove)
 	testObjectAction(t, func() error { return w.Resize(1, 2) }, w.object, wrt, "{\"name\":\""+EventNameWindowCmdResize+"\",\"targetID\":\""+w.id+"\",\"windowOptions\":{\"height\":2,\"width\":1}}\n", EventNameWindowEventResize)
+	bounds, err := w.Bounds()
```
Comment (asticode) [inline window_test.go:68]: Could you update tests for functions that will update bounds as well?

- make sure to return a `bounds` key in the terminated event
- make sure to test `Bounds()` return the proper values

**Review by asticode — CHANGES_REQUESTED** [review]

File: window_test.go (hunk 2 above)
Comment (maddie) [reply window_test.go:68]: > Could you update tests for functions that will update bounds as well?
> 
> * make sure to return a `bounds` key in the terminated event
> * make sure to test `Bounds()` return the proper values

I can add a test for `Move` function, but for others like `Maximize`, `ResizeContent`, I'm not sure of how to determine its returned bounds in the events.

File: window.go (hunk 1 above)
Comment (maddie) [reply window.go:274]: Done

File: window_test.go (hunk 2 above)
Comment (asticode) [reply window_test.go:68]: for other actions like `Maximize`, tests simulate what's returned by `astilectron` therefore all you need to do is add a `bounds` key with proper value

Comment (maddie) [reply window_test.go:68]: I can't find where to attach the `bounds` in the returned event, do I have to modify `testObjectAction` to allow passing in a customized return event?

Comment (asticode) [reply window_test.go:68]: I agree you have to modify `testObjectAction` but I would change its signature to this instead:

```go
func testObjectAction(t *testing.T, fn func() error, o *object, wrt *mockedWriter, sentEvent, eventNameDone string, receivedEvent *Event)
```

You'll have to add `, nil)` for other tests but that'd be more explicit.

Comment (maddie) [reply window_test.go:68]: Done

**Comment by asticode** [comment]
Cheers

**Pull request merged**